### DISCUSSION TITLE
suggestion: more distinguishable icon on macOS

### DISCUSSION THREAD

**Discussion by marco-m-pix4d** [gulbanana/gg#14] [Ideas]
Hello,
thanks for gg!

On macOS, when using the task switcher (cmd-tab), the following list appears:
![gg](https://github.com/user-attachments/assets/6e283f8f-b75b-4b4f-ae38-6873aa7e18ea)

The gg icon takes all the space, while the other icons (in the screenshot: Preview and Firefox) have a transparent border. This transparent border is useful to visually distinguish which program is currently selected.

In the screenshot, Firefox is currently selected. If I hit TAB once more, I would select gg, but it would be almost impossible to visually distinguish it, since the gray border would be renderered almost completely over the icon.

thanks!

**Comment by Jollywatt** [comment]
For what it's worth, I used http://icon.kitchen to make a bearable app icon. Something like [this](https://icon.kitchen/i/H4sIAAAAAAAAAyWQsW7DMAxE%2F4VdPcR23AJe07Rrh2xFUdASxQilLEOWAxSB%2Fz2UMpF4R94RvMMNZaMVxjtYTH%2BXKwWCMaeNGnB8Er9gykVdSQtYcrhJhga8ibMCgyK%2FgRIT7A1MfPlfdB9MlJh0auLPhNbTXD0qbVV%2BGfrX94%2BzDlTUFdRP3fGNFOHMoh7HoRqeqpPqLXWH%2FqC647NzZLIeDSR0w1y2HH%2BhtX7mEpTjAmM7NJA8XzW6tFPMOYZnL%2BQq3TUiRLtJ%2BcA3BDRxhZ%2F9AYMgvXEWAQAA), for example. Download the icon file, open the info panel for the gg app, and drag your file onto the icon in the top left.

**Comment by djeikyb** [comment]
also  fwiw (not weighing in on OP's point), i love this bonkers creepy neon-glowing eye of an icon

**Reply by ilyagr** [reply]
FR: the eye should randomly blink approximately once in a few hours. :)

This would depend on https://github.com/tauri-apps/tauri/issues/2358.

**Comment by gulbanana** [comment]
agreed, higher-res and transparency would be nice

**Comment by lgarron** [comment]
I wanted my `gg` dock icon to use the `jj` icon, so I made a version sized to a typical macOS icon (at higher resolutions)[^1]:

![gg icon (1024w)](https://github.com/user-attachments/assets/84d1f0f9-7ff6-4827-8ad8-cc73c6a6a6cf)

I've been using it for a while now, and I'm pretty happy with how it looks!
![clipboard](https://github.com/user-attachments/assets/9e9b8272-b2f2-4d79-a300-6cdf92e69a3e)

[^1]: In fact, the `jj` project now has a vector icon specifically because I wanted a high-resolution icon for `gg`: https://github.com/jj-vcs/jj/pull/5958

**Reply by lgarron** [reply]
An updated version of the icon for macOS 26 Tahoe:

<img width="2048" height="2048" alt="jj-logo-v0 1 3 app-icon" src="https://github.com/user-attachments/assets/25627d55-10de-4792-99de-dfab21532dbc" />

**Reply by lgarron** [reply]
@gulbanana: It seems that https://github.com/gulbanana/gg/commit/ca63423f0a846c06500154569ece317da7798edc now overrides any icon I set for the app. It launches with the icon I've set, then switches to the built-in one in a fraction of a second. This also happens when I launch the app normally (not via `gg` CLI).

Any chance I could ask for an opt-out, override, or any other way to set the icon above?

**Reply by gulbanana** [reply]
i’m tempted to just switch to your icon, as long as the jj project doesn’t object. either way, we can add a setting to disable the icon-forcing effect, or apply it only when using the non-.app build

**Reply by lgarron** [reply]
> i’m tempted to just switch to your icon, as long as the jj project doesn’t object.

I'd certainly love that! I've asked at https://github.com/jj-vcs/jj/issues/8408

**Comment by gulbanana** [comment]
The gg icon correctly takes on macos squircle shapes if you use the app build (.dmg, .tar.gz). Previously this had the downside of blocking the shell, but I've fixed that. If you were using `cargo install` on MacOS, I recommend switching to `brew install --cask gg`.
